iib/workers/tasks/utils.py

```diff
@@ -147,12 +155,13 @@ def set_registry_token(token, container_image):
 
 
 @retry(wait_on=IIBError, logger=log)
-def skopeo_inspect(*args, return_json=True):
+def skopeo_inspect(*args, return_json=True, use_cache=True):
```

Comment:
I would prefer if you used [cache_on_arguments](https://dogpilecache.sqlalchemy.org/en/latest/api.html#dogpile.cache.region.CacheRegion.cache_on_arguments) and then passed in function for the `should_cache_fn` keyword argument that will only return `True` if the pull spec passed into `skopeo_inspect` uses a digest. Thent he caller does not have to think about if using the cache is necessary.

Reply:
Hi @mprahl , I could not do it so I am not using `cache_on_argument`. Since `should_cache_fn` does not have data about pull spec input. There is passed only the result of `skopeo_inspect` function. I am caching only data when the image was specified with digest.

Could you please check the code if it is OK, or if you know about different solution please suggest one. 
Thank you!

Reply:
I see what you mean. `should_cache_fn` does something different than I thought. Perhaps a decorator that does this would be cleaner. The decorator could accept a function which analyzes the arguments to determine if the cache should be used.

Reply:
Thanks for this suggestion, I've created a decorator so the `skopeo_inspect` function remains untouched. 
Could you check it, please.